### ISSUE TITLE
Add SwinV2 in TorchVision

### ISSUE THREAD

**Issue by datumbox** [pytorch/vision#6242]
### 🚀 The feature

Swin Transforms were added to TorchVision on #5491 thanks to the work of @xiaohu2015 and @jdsgomes. At that point, the official code of [SwinV2](https://arxiv.org/abs/2111.09883) was not released so we decided to postpone its addition. Later @YosuaMichael has refactored the class to make it compatible with 3d use-cases at #6088. With the code of V2 now [available](https://github.com/microsoft/Swin-Transformer/blob/main/models/swin_transformer_v2.py), we should consider adding it to TorchVision.

Here is what needs to be done:
- Examine if it's possible to extend the existing [Swin Transformer](https://github.com/pytorch/vision/blob/main/torchvision/models/swin_transformer.py) class to support both V1 and V2 models. An example of such an extension was recently done for [EfficientNets](https://github.com/pytorch/vision/blob/main/torchvision/models/efficientnet.py). On the other hand, if the implementations are substantially different, we can follow the approach we took for MobileNets [v2](https://github.com/pytorch/vision/blob/main/torchvision/models/mobilenetv2.py) and [v3](https://github.com/pytorch/vision/blob/main/torchvision/models/mobilenetv3.py) and use separate classes.
- Provide an implementation that uses TorchVision building blocks and idioms. See #5319 for pointers.
- Provide proof in form of pre-trained weights for the smallest variant that reproduces the accuracy of the paper.

### Motivation, pitch

We should keep TorchVision fresh by offering the latest influential architectures to support the work of ML researchers and practitioners.

cc @datumbox

**Comment by datumbox** [comment]
@ain-soph thanks for the PR! Please comment on this ticket to be able to assign it to you. Github won't let me do it. :(

**Comment by ain-soph** [comment]
Sure!
